### PR TITLE
fix(#1085): Propagate Impersonate attributes

### PR DESCRIPTION
**Pull request by lordofthejars**
#1085

### CODE REVIEW

**Comment by piyush1594** [comment]
ok to test

**Comment by piyush1594** [comment]
@lulf @ibuziuk Can you please take a look?

**Comment by lulf** [comment]
LGTM. Consider adding a test to ConfigTest.

**Comment by lordofthejars** [comment]
@lulf Do you think it is worth to implement these parameters to be set as system properties as well? Or just propagating?

**Comment by lulf** [comment]
@lordofthejars Agreed, it would make it consistent with how the other settings are handled.

**Comment by ibuziuk** [comment]
LGTM

**Comment by lordofthejars** [comment]
@lulf Added tests and sys prop support. I have not implemented system properties for extra parameters. My point is that extra parameters field is a map, and usually you have two options, the first one is creating a prefixed system property where last part of key is the key and that value the map:

i.e:

```
kubernetes.impersonate.extras.key1=value
kubernetes.impersonate.extras.key2=vale
```

But you need to scan for all preferences and env vars and iterate.

The other option is to parse a string: `kubernetes.impersonate.extras=key1=value, key2=value`

Based on my experience is that if it is not totally required, the best way is to avoid configuring maps but what do you think about this concrete case?

File: kubernetes-client/src/test/java/io/fabric8/kubernetes/client/ConfigTest.java [hunk 1]
```diff
@@ -17,13 +17,17 @@
 package io.fabric8.kubernetes.client;
 
 import io.fabric8.kubernetes.client.utils.Serialization;
+import io.fabric8.openshift.clnt.v3_1.DefaultOpenShiftClient;
```
Comment (piyush1594) [inline kubernetes-client/src/test/java/io/fabric8/kubernetes/client/ConfigTest.java]: You need to fix this import @lordofthejars

**Comment by piyush1594** [comment]
@lordofthejars Please add changelog.md also

File: kubernetes-client/src/test/java/io/fabric8/kubernetes/client/ConfigTest.java (hunk 1 above)
Comment (lordofthejars) [reply kubernetes-client/src/test/java/io/fabric8/kubernetes/client/ConfigTest.java]: Done

**Comment by lordofthejars** [comment]
@piyush1594 Done, when everyone approves the PR I will squash everything.

**Comment by lulf** [comment]
@lordofthejars LGTM

**Comment by piyush1594** [comment]
Thanks @lordofthejars

**Pull request merged**